src/model/test/FloorspaceReverseTranslator_GTest.cpp

```diff
@@ -1389,3 +1406,480 @@ TEST_F(ModelFixture, FloorspaceReverseTranslator_FloorplanJS_Site_ClimateZones_4
+// Test to verify all floor surfaces on Story 2 are "Surface"
+// USes the same test file as 4203
+TEST_F(ModelFixture, FloorspaceReverseTranslator_Issue_4204) {
+
+  ThreeJSReverseTranslator rt;
+
+  openstudio::path p = resourcesPath() / toPath("utilities/Geometry/issue-4203.json");
+  ASSERT_TRUE(exists(p));
+
+  boost::optional<FloorplanJS> floorPlan = FloorplanJS::load(toString(p));
+  ASSERT_TRUE(floorPlan);
+
+  // not triangulated, for model transport/translation
+  ThreeScene scene = floorPlan->toThreeScene(true);
+
+  boost::optional<Model> model = rt.modelFromThreeJS(scene);
+  ASSERT_TRUE(model);
+  ShiftVertices(*model);
+  model->save(resourcesPath() / toPath("utilities/Geometry/issue_4204-threejs.osm"), true);
+
+  auto story2 = model->getConcreteModelObjectByName<BuildingStory>("Story 2");
+  ASSERT_TRUE(story2);
+  auto spaces = story2->spaces();
+  for (auto& space : spaces) {
+    std::string name = space.nameString();
+    const auto& surfaces = space.surfaces();
+    for (const auto& surface : surfaces) {
+      if (surface.surfaceType() == "Floor") {
+        EXPECT_EQ(surface.outsideBoundaryCondition(), "Surface") << surface.nameString();
+      }
+    }
+  }
+
+  FloorspaceReverseTranslator frt;
+  boost::optional<Model> model1 = frt.modelFromFloorspace(toString(p));
+  ASSERT_TRUE(model1);
+  ShiftVertices(*model1);
+  model1->save(resourcesPath() / toPath("utilities/Geometry/issue_4204-direct.osm"), true);
+
+  story2 = model1->getConcreteModelObjectByName<BuildingStory>("Story 2");
+  ASSERT_TRUE(story2);
+  auto spaces1 = story2->spaces();
+  for (auto& space1 : spaces1) {
+    auto surfaces1 = space1.surfaces();
+    for (const auto& surface : surfaces1) {
+      if (surface.surfaceType() == "Floor") {
+        EXPECT_EQ(surface.outsideBoundaryCondition(), "Surface") << surface.nameString();
+      }
+    }
+  }
+
+  CompareTwoModels(*model1, *model);
+}
+
+TEST_F(ModelFixture, FloorspaceReverseTranslator_Issue_4237) {
+
+  ThreeJSReverseTranslator rt;
+
+  openstudio::path p = resourcesPath() / toPath("utilities/Geometry/issue_4237.json");
+  ASSERT_TRUE(exists(p));
+
+  boost::optional<FloorplanJS> floorPlan = FloorplanJS::load(toString(p));
+  ASSERT_TRUE(floorPlan);
+
+  // not triangulated, for model transport/translation
+  ThreeScene scene = floorPlan->toThreeScene(true);
+
+  boost::optional<Model> model = rt.modelFromThreeJS(scene);
+  ASSERT_TRUE(model);
+  model->save(resourcesPath() / toPath("utilities/Geometry/issue_4237-threejs.osm"), true);
+
+  FloorspaceReverseTranslator frt;
+  boost::optional<Model> model1 = frt.modelFromFloorspace(toString(p));
+  ASSERT_TRUE(model1);
+
+  model1->save(resourcesPath() / toPath("utilities/Geometry/issue_4237-direct.osm"), true);
+
+  CompareTwoModels(*model1, *model);
+}
+
+TEST_F(ModelFixture, FloorspaceReverseTranslator_Issue_4776) {
+
+  ThreeJSReverseTranslator rt;
+
+  openstudio::path p = resourcesPath() / toPath("utilities/Geometry/issue-4203.json");
```

Comment:
Is this normal the json and issue aren't matching?

Reply:
@jmarrec The test is re-using the same json file.